scripts/pre-commit/check_version.py

```diff
@@ -0,0 +1,176 @@
+
+
+def get_cmd_output(cmd: typing.List[str]) -> str:
+    """
+    Get the subprocess output for the command to be run
+    :param cmd: command parameters as list
+    :return: Command output
+    """
+    try:
+        return subprocess.check_output(cmd).decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        raise Exception(f"Could not retrieve command output: {e}")
+
+
+def get_all_modified_files() -> typing.List[pathlib.Path]:
+    """
+    Retrieve the list of files updated from the upstream branch
+    """
+    try:
+        files_updated = get_cmd_output(GitCommand.MODIFIED_FILES)
+        return [pathlib.Path(path) for path in files_updated.splitlines()]
+    except subprocess.CalledProcessError as e:
+        raise Exception(f"Could not retrieve git diff: {e}")
+
+
+def get_modified_pipelines() -> typing.List[pathlib.Path]:
+    """
+    Get the list of changed pipelines and metadata store scripts
+    """
+    modified_files = []
+    for file_path in get_all_modified_files():
+        parent = file_path.parent.name
+        filetype = file_path.suffix
+        if (
+            (parent.startswith(TEMPLATES_JSON_PATH_PREFIX) and filetype == ".json")
+            or (parent.startswith(METADATA_STORE_PATH) and filetype == ".sql")
+        ):
+            modified_files.append(file_path)
+
+    return modified_files
+
+
+def get_version_from_version_md(version_md: str) -> "packaging.version.Version":
+    """
+    Previous version of a file
+    :return: previous version
+    """
+    try:
+        return parse_version(version_md)
+    except ValueError as e:
+        raise VersionParsingException(f"Invalid version number: {str(e)}")
+    except Exception as e:
+        logging.warning(f"Cannot find previous version: {e}")
+
+
+def get_previous_version() -> str:
+    """
+    Parse version from the previous version of the file
+    """
+    version_md = get_cmd_output(
+        GitCommand.FILE_CONTENT + [f"origin/main:{VERSION_FILE}"]
+    ).strip()
+    return get_version_from_version_md(version_md)
+
+
+def get_current_version() -> str:
+    """
+    Parse version from the current version of the file
+    """
+    with open(VERSION_FILE, "r") as f:
+        version_md = f.read().strip()
+    return get_version_from_version_md(version_md)
+
+
+def get_commit_message() -> str:
+    """
+    Get the commit message
+    """
+    return get_cmd_output(GitCommand.COMMIT_MESSAGE).strip()
+
+
+def validate() -> None:
+    """
+    Validate current repo version against the previous version
+    """
+    if not pathlib.Path(VERSION_FILE).exists():
+        raise FileNotFoundException(f"Error: {VERSION_FILE} file does not exist.")
+
+    current_version = get_current_version()
+    previous_version = get_previous_version()
+
+    if current_version <= previous_version:
+        raise InvalidVersionException(
+            f"file://{pathlib.Path(VERSION_FILE).absolute()} {previous_version} -->"
```

Comment:
This will work too, 
```suggestion
            f"{pathlib.Path(VERSION_FILE).absolute().as_uri()} {previous_version} -->"
```

Also construct the path in the beginning of the method to make it reusable. 
`path = pathlib.Path(VERSION_FILE)`

Reply:
Done.